### PR TITLE
Adding pascalCase transform to be used by brand

### PR DESCRIPTION
**Pull request by lukasoppermann**
This PR adds a pascalCase transformer

### CODE REVIEW

File: .changeset/dry-walls-marry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/primitives': minor
```
Comment (rezrah) [inline .changeset/dry-walls-marry.md:2]: Just curious how you determine the version. Why minor vs patch?

File: src/utilities/toPascalCase.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+export const toPascalCase = (string: string | string[]) => {
+  if (!Array.isArray(string)) {
+    string = [string]
+  }
+  // match unsupported characters
+  const regex = /[^a-zA-Z0-9]+/g
+  // replace any non-letter and non-number character and split into word array
+  const stringArray = string
+    .filter(part => part !== '@')
+    .filter(Boolean)
```
Comment (rezrah) [inline src/utilities/toPascalCase.ts:12]: Nit, but you could combine these

File: src/utilities/toPascalCase.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    .split(' ')
+  return (
+    stringArray
+      // remove undefined if exists
+      .filter((part: unknown): part is string => typeof part === 'string')
+      // ucFirst all but first part
+      .map((part: string) => {
+        return upperCaseFirstCharacter(part)
+      })
+      .join('')
```
Comment (rezrah) [inline src/utilities/toPascalCase.ts:24]: Another nit (not blocking) but you could do all array manipulation inside the function body and just return the final output in the return to make this a bit clearer. Right now it seems like there's the same things happening in both the return and function body.

**Review by rezrah — APPROVED** [review]
Left some minor, non-blocking feedback.

Thanks for doing this @lukasoppermann 🙏

File: .changeset/dry-walls-marry.md (hunk 1 above)
Comment (lukasoppermann) [reply .changeset/dry-walls-marry.md:2]: @rezrah I figure we are adding a feature here, so minor is more appropriate. Let me know if you have some good guidelines as to when to choose which version.

File: src/utilities/toPascalCase.ts (hunk 2 above)
Comment (lukasoppermann) [reply src/utilities/toPascalCase.ts:12]: Yes, but I feel it is easier to read this way and performance is not really a bottle-neck.

You mean like this, right?

```js
 .filter(part => part !== '@' && Boolean(part))
````

File: src/utilities/toPascalCase.ts (hunk 3 above)
Comment (lukasoppermann) [reply src/utilities/toPascalCase.ts:24]: I thought about this as well. Will look into it. 👍

File: .changeset/dry-walls-marry.md (hunk 1 above)
Comment (rezrah) [reply .changeset/dry-walls-marry.md:2]: Yeah that makes sense. Thanks.

File: src/utilities/toPascalCase.ts (hunk 2 above)
Comment (lukasoppermann) [reply src/utilities/toPascalCase.ts]: Actually ended up refactoring this a bit

**Pull request merged**